server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -202,7 +203,7 @@ private boolean mergeRule(RulesDefinition.Rule def, RuleDto dto) {
     if (mergeDescription(def, dto)) {
       changed = true;
     }
-    if (!dto.getSystemTags().containsAll(def.tags())) {
+    if (!difference(dto.getSystemTags(), def.tags()).isEmpty()) {
```

Comment:
Indeed, it should have failed, but another merge of tags is done line 346 !
I'm fixing this